src/assets/src/utils.ts

```diff
@@ -18,10 +19,20 @@ export const redirectToSearch = (term: string) => {
     location.href = `/search/${term}/?redirected=true`;
 }
 
-export const validateUniqname = (uniqname: string) => {
+const invalidUniqnameMessage = (uniqname: string) =>
+    uniqname + " is not a valid user. Please make sure the uniqname is correct, and that they have logged onto Remote Office Hours Queue at least once."
+
+export const validateAndFetchUserByUniqname = async (uniqname: string) => {
```

Comment:
I appreciate the move of `invalidUniqnameMessage` out of `common.tsx`. However, it feels a little like we're smushing stuff together that shouldn't necessarily be combined. The call of `getUserByUniqname` and handling of errors feels like it could be its own function in `api.ts` (the error handling could even get moved into `getUserByUniqname`). Also, I'm probably biased because of my previous work on #143, but the validation piece could be left separate from the fetching. I dunno, `validateAndFetchUserByUniqname` just feels like a mouthful, which says to me it may make sense conceptually to keep them apart.

All of that said, this should work, and it doesn't necessarily need to be changed now.

Reply:
My goal here was to avoid putting validation bits in `api` which has previously been divorced from input validation, but I agree that this piece is a bit odd standing out in `utils` as the only such "validate and call API with some error handling" type of function. This can be revisited in your input validation PR.